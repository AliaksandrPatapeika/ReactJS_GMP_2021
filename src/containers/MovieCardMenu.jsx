import '../less/containers/MovieCardMenu.less';

import PropTypes from 'prop-types';
import React from 'react';

const MovieCardMenu = ({toggleMenu}) => (
  <div className="movieCardMenuContainer">
    <button type="button" className="menuItem closeButton" onClick={toggleMenu}>&#x2715;</button>
    <nav>
      <ul>
        <li role="menuitem" className="menuItem">Edit</li>
        <li role="menuitem" className="menuItem">Delete</li>
      </ul>
    </nav>
  </div>
);

MovieCardMenu.propTypes = {
  toggleMenu: PropTypes.func.isRequired
};

export default MovieCardMenu;
