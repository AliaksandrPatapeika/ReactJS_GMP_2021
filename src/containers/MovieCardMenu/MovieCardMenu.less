@import "../../less/variables";

.movieCardMenuContainer {
	position: absolute;
	right: 8px;
	top: 8px;
	display: flex;
	flex-direction: column;
	width: 160px;
	border-radius: 4px;
	background-color: @black;

	li:last-of-type {
		margin-bottom: 10px;
	}

	.menuItem {
		cursor: pointer;
		padding: 4px 16px;
		color: @white;
		font-size: 18px;
		font-family: @NetflixSans-Regular;

		&:hover {
			background-color: @red;
		}
	}

	.closeButton {
		align-self: flex-end;
		margin: 6px 6px 0 0;
		padding: 3px 7px;
		border-radius: 2px;
		background-color: @black;
	}
}