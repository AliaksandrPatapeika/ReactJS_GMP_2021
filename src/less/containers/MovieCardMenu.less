@import "../variables";

.movieCardMenuContainer {
	display: flex;
	flex-direction: column;
	background-color: @black;
	position: absolute;
	border-radius: 4px;
	right: 8px;
	top: 8px;
	width: 160px;

	li:last-of-type {
		margin-bottom: 10px;
	}

	.menuItem {
		font-family: @NetflixSans-Regular;
		font-size: 18px;
		color: @white;
		padding: 4px 16px;
		cursor: pointer;

		&:hover {
			background-color: @red;
		}
	}

	.closeButton {
		border-radius: 2px;
		background-color: @black;
		align-self: flex-end;
		margin: 6px 6px 0 0;
		padding: 3px 7px;
	}
}