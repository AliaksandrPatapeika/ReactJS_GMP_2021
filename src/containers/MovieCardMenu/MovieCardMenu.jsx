import './MovieCardMenu.less';

import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import PropTypes from 'prop-types';
import React from 'react';

import Button from '../../components/Button';

const MovieCardMenu = ({closeSubMenu, showEditWindow, showDeleteWindow}) => (
  <ClickAwayListener onClickAway={closeSubMenu}>
    <div className="movieCardMenuContainer">
      <Button className="closeButton" onClick={closeSubMenu}>
        <i className="fa fa-times" aria-hidden="true" />
      </Button>
      <nav>
        <ul>
          <li
            role="menuitem"
            tabIndex="0"
            className="menuItem"
            onClick={showEditWindow}
            onKeyPress={showEditWindow}
          >
            Edit
          </li>
          <li
            role="menuitem"
            tabIndex="0"
            className="menuItem"
            onClick={showDeleteWindow}
            onKeyPress={showDeleteWindow}
          >
            Delete
          </li>
        </ul>
      </nav>
    </div>
  </ClickAwayListener>
);

MovieCardMenu.propTypes = {
  closeSubMenu: PropTypes.func.isRequired,
  showEditWindow: PropTypes.func.isRequired,
  showDeleteWindow: PropTypes.func.isRequired
};

export default MovieCardMenu;
