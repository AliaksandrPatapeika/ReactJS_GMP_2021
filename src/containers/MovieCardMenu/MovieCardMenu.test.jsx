import React from 'react';
import renderer from 'react-test-renderer';

import MovieCardMenu from './MovieCardMenu';

describe('MovieCardMenu', () => {
  test('renders correctly', () => {
    const movieCardMenu = renderer
      .create(<MovieCardMenu
        closeSubMenu={Function}
        showEditWindow={Function}
        showDeleteWindow={Function}
      />)
      .toJSON();

    expect(movieCardMenu).toMatchSnapshot();
  });
});
