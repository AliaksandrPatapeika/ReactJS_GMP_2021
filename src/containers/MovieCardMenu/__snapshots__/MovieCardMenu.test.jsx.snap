// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardMenu renders correctly 1`] = `
<div
  className="movieCardMenuContainer"
  onClick={[Function]}
  onTouchEnd={[Function]}
>
  <button
    className="closeButton"
    disabled=""
    onClick={[Function]}
    type="button"
  >
    
    <i
      aria-hidden="true"
      className="fa fa-times"
    />
  </button>
  <nav>
    <ul>
      <li
        className="menuItem"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="menuitem"
        tabIndex="0"
      >
        Edit
      </li>
      <li
        className="menuItem"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="menuitem"
        tabIndex="0"
      >
        Delete
      </li>
    </ul>
  </nav>
</div>
`;
